### PR TITLE
fix(providers): pin both Daytona variants to us-west-2 and make the pin reach the API

### PR DESCRIPTION
**Pull request by dbworku**
**Bench-matrix green — every suite×provider cell measuring or honestly gapped**
Fixes the 10 root causes behind the failures and silent metric holes of runs [29799034615](https://github.com/starslingdev/hpc-sandbox-benchmarks/actions/runs/29799034615) and [29850811070](https://github.com/starslingdev/hpc-sandbox-benchmarks/actions/runs/29850811070). #229–#232 have merged; the remaining fixes are **parallel PRs based directly on `main`** — reviewable and mergeable in any order, except #238, which stacks on #237:
- #233 — leaderboard: no fabricated missing-rows for never-dispatched providers
- #234 — realworld: per-task timeout + cgroup-v2 memory containment
- #235 — harness: failed gap on sandbox-creation failure
- #236 — harness: detached log read-back + collect retries
- #237 — bench: loud PTS leaf guards; stream + pybench measure again
- #238 — templates: pgbench bake fix (pkg-config), payload assertion, portable fio, v4 *(stacked on #237)*
- #239 — fast-cli: settle gated on upload-phase stability
- #240 — providers: Daytona region pin actually reaches the API; both variants us-west-2
- #241 — ci: dataset-PR PAT fallback; e2b disk-skip notes

↑ **This PR is #240 — independent: based on `main`, no merge-order dependency on the others.**

---
All 9 daytona-container cells of run 29850811070 died at creation:
'Snapshot <org>-toolchain-v3-container is not available in region ZEN5-VM'.
Two stacked defects:

1. The adapter's region pin was dead code end-to-end (offline-replayed
   with the real wrapper + SDK and a stubbed transport): @computesdk/daytona
   never passes createOptions.target into the native client, and the
   Daytona SDK's create() ignores createParams.target — it honors ONLY the
   client-level target (constructor config or the DAYTONA_TARGET env
   fallback). Container jobs POSTed target:'' and fell to the account
   default region; daytona-vm worked only because its job env happened to
   feed the SDK's env fallback. daytonaClientTarget() now pins
   DAYTONA_TARGET around the wrapped create (the e2b-root patchableManager
   precedent; restore-in-finally; race-free — each CI job runs exactly one
   provider), and the dead createOption is deleted with a regression guard.
2. The container variant's default region pointed at a retired region.
   Both variants now default to us-west-2 (config + the three workflow
   fallbacks), where the container snapshot is imported at bake time.

The replay test asserts the POSTed body carries the config target for both
variants, with env restore semantics on success and rejection. Maintainer
action (PR description): the privileged-environment secret
DAYTONA_CONTAINER_TARGET still carries the retired region and overrides
any code default — update it to us-west-2 or delete it.

---
## Summary by cubic
Pins both Daytona providers to us-west-2 and ensures the region pin reaches the API across create and teardown paths. Prevents wrong‑region boots and leaked sandboxes.

- **Bug Fixes**
  - Pin via client-level target: wrap `@computesdk/daytona` with `daytonaClientTarget` to set/restore `DAYTONA_TARGET` for create, getById, list, and destroy; honor per-create `target`; patch per-instance with a shape guard.
  - Remove dead `createOptions.target`; add offline tests that stub the SDK transport to verify POSTed `target`, env restore (set/empty/unset), teardown-path pinning, and instance-level patching.
  - Default both `daytona-vm` and `daytona-container` to us-west-2 in config and workflow fallbacks (`bench-smoke`, `bench-suite`, `toolchain-image`); update bake/validate code and tests.

- **Migration**
  - Update the secret `DAYTONA_CONTAINER_TARGET` to `us-west-2` (or delete it) so it doesn’t override the new default.

<sup>Written for commit d08eb6155188d4353fa6db601904b3d3fc873c17. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/starslingdev/hpc-sandbox-benchmarks/pull/240?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

---
**Maintainer action required (code defaults cannot beat a set secret):**
- [ ] The `privileged` environment secret `DAYTONA_CONTAINER_TARGET` still carries the retired region that killed all 9 daytona-container cells — update it to `us-west-2` or delete it (Settings → Environments → privileged).
- [ ] Policy question: promote daytona-container to a *required* provider in `toolchain-image.yml`'s bake gate? Its bake-verify would have caught this at v3 bake time, but non-required providers only warn.

### CODE REVIEW

**Comment by dbworku** [comment]
* **#240** <a href="https://app.graphite.com/github/pr/starslingdev/hpc-sandbox-benchmarks/240?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.com/github/pr/starslingdev/hpc-sandbox-benchmarks/240?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add the label <kbd>merge-ready</kbd> to this PR to add it to the merge queue.

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.com/invite/github/starslingdev?ref=merge-queue-instructions-in-stack-comment&prId=240).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: .github/workflows/toolchain-image.yml [hunk 1]
```diff
@@ -324,13 +324,13 @@ jobs:
...
-          # Each Daytona variant bakes in its own region, pinned by a variant-specific secret so
+          # Both Daytona variants bake in us-west-2, pinned by a variant-specific secret so
           # operators can retarget without editing the workflow: daytona-vm reads DAYTONA_TARGET
           # (LINUX_VM snapshot, us-west-2 default), daytona-container reads DAYTONA_CONTAINER_TARGET
-          # (`us` default). Both default so an unsynced secret can't leave a snapshot in the account's
+          # (us-west-2 default). Both default so an unsynced secret can't leave a snapshot in the account's
           # default region and fail the bake at snapshot creation.
           DAYTONA_TARGET: ${{ matrix.provider == 'daytona-vm' && (secrets.DAYTONA_TARGET || 'us-west-2') || '' }}
-          DAYTONA_CONTAINER_TARGET: ${{ matrix.provider == 'daytona-container' && (secrets.DAYTONA_CONTAINER_TARGET || 'us') || '' }}
+          DAYTONA_CONTAINER_TARGET: ${{ matrix.provider == 'daytona-container' && (secrets.DAYTONA_CONTAINER_TARGET || 'us-west-2') || '' }}
```
[the remark this answers is not in the document]
Comment (dbworku) [reply .github/workflows/toolchain-image.yml:333]: Correct, and deliberate: a set secret must override code defaults — that is the operators' retarget knob, and the workflow fallback only covers the unset/empty case. Updating/deleting the stale `DAYTONA_CONTAINER_TARGET` secret is the first maintainer-action checkbox in this PR's description; code cannot unset an environment secret.

File: packages/providers/src/lib/daytona-target.ts [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		} finally {
+			// delete, not `= undefined` — assigning undefined to process.env stringifies to "undefined".
+			if (previous === undefined) {
+				delete process.env.DAYTONA_TARGET;
+			} else {
+				process.env.DAYTONA_TARGET = previous;
+			}
+		}
+	};
+	manager.methods = { ...manager.methods, create: pinnedCreate };
```
Comment (cosmic-emerald) [inline packages/providers/src/lib/daytona-target.ts]: Should the target wrapper also patch `destroy` and other methods that construct a fresh Daytona client?

**Review by cosmic-emerald — APPROVED** [review]

File: packages/providers/src/lib/daytona-target.ts (hunk 2 above)
Comment (dbworku) [reply packages/providers/src/lib/daytona-target.ts]: Good catch — yes, and it mattered for teardown. `getById`, `list`, and `destroy` each build a fresh `new Daytona({ apiKey })` whose region resolves from the same `DAYTONA_TARGET` fallback as `create`, but only `create` ran under the pin. So a container CI job (where `DAYTONA_TARGET` materializes set-but-empty) would create in `us-west-2` yet run `sandbox.destroy()` against the account default region — missing the sandbox and leaking it.

Fixed in `3495110`: `daytonaClientTarget` now wraps every client-constructing sandbox method through a shared `withTargetPin` helper (finally-restore + the `"undefined"`-leak guard preserved). `create` keeps its per-call target override (validate.ts's candidate region); `getById`/`list`/`destroy` key off the adapter's configured target. The shape guard checks all four now, so an upstream rename fails loudly, and there are fake-provider tests proving the teardown path runs under the pin with set/empty/unset restore semantics. The `runCommand`/`getInfo`/`getUrl`/`filesystem.*` methods act on an already-constructed sandbox object, so they build no client and stay stock.

---
_Generated by [Claude Code](https://claude.ai/code)_
